src/ni/datastore/_auth.py

```diff
@@ -0,0 +1,442 @@
+import base64
+import collections
+import json
+import logging
+import os
+import platform
+import shutil
+import subprocess
+import time
+from pathlib import Path
+from typing import Optional
+
+import grpc
+
+from ni_grpc_extensions.channelpool import GrpcChannelPool
+
+_logger = logging.getLogger(__name__)
+
+
+class _ClientCallDetails(
+    collections.namedtuple(
+        "_ClientCallDetails",
+        (
+            "method",
+            "timeout",
+            "metadata",
+            "credentials",
+            "wait_for_ready",
+            "compression",
+        ),
+    ),
+    grpc.ClientCallDetails,
+):
+    pass
+
+
+# Create an interceptor that adds the Authorization header
+class _AuthInterceptor(grpc.UnaryUnaryClientInterceptor):
+    def __init__(self, token):
```

Comment:
long term, we don't want to pass the token in here - we want to pass in the thing that can _get_ the token (otherwise it goes stale). Maybe just pass in the JwtTokenProvider, rather than passing in the token itself?